python/ray/tests/test_autoscaler_yaml.py

```diff
@@ -275,8 +281,80 @@ def testMaxWorkerDefault(self):
             prepared_node_types["worker_node_max_specified"][
                 "max_workers"] == 3
         # Max workers auto-filled with default cluster-wide value of 2.
-        assert prepared_node_types["worker_node_max_unspecified"][
-            "max_workers"] == 2
+        assert prepared_config["max_workers"] ==\
+            prepared_node_types["worker_node_max_unspecified"]["max_workers"]\
+            == 2
+        # Head type max workers filled to 0
+        assert prepared_config["available_node_types"]["head_node"][
+            "max_workers"] == 0
+
+    def testGetDefaultHeadWorker(self):
+        """
+        Test the functions get_default_head and get_default_workers.
+        """
+        providers = ["aws", "gcp", "azure"]
+        for provider in providers:
+            path = os.path.join(RAY_PATH, "autoscaler", provider,
+                                "defaults.yaml")
+            default_config = yaml.safe_load(open(path).read())
+            assert get_default_head(default_config) == default_config[
+                "available_node_types"]["ray.head.default"]["node_config"]
+            assert get_default_workers(default_config) == default_config[
+                "available_node_types"]["ray.worker.small"]["node_config"]
+
+    def testFillEdgeLegacyConfigs(self):
+        # Test edge cases: legacy configs which specify workers but not head
+        # or vice-versa.
+        no_head = load_test_config("test_no_head.yaml")
+        aws_defaults = _get_default_config(no_head["provider"])
+        head_prepared = prepare_config(no_head)
+        assert head_prepared["available_node_types"][
+            "ray-legacy-head-node-type"]["node_config"] ==\
+            head_prepared["head_node"] ==\
+            aws_defaults["available_node_types"]["ray.head.default"][
+                "node_config"]
+
+        no_workers = load_test_config("test_no_workers.yaml")
+        workers_prepared = prepare_config(no_workers)
+        assert workers_prepared["available_node_types"][
+            "ray-legacy-worker-node-type"]["node_config"] ==\
+            workers_prepared["worker_nodes"] ==\
+            aws_defaults["available_node_types"]["ray.worker.small"][
+                "node_config"]
+
+    def testExampleFull(self):
```

Comment:
can we add a test for example minimal?

Reply:
What behavior do you think we should test for example-minimal?